osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs

```diff
@@ -33,13 +33,15 @@ public class OsuModRelax : ModRelax, IUpdatableByPlayfield, IApplicableToDrawabl
 
         private ReplayState<OsuAction> state;
         private double lastStateChangeTime;
+        private double beatmapLength;
 
         private bool hasReplay;
 
         public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
         {
             // grab the input manager for future use.
             osuInputManager = (OsuInputManager)drawableRuleset.KeyBindingInputManager;
+            beatmapLength = drawableRuleset.Beatmap.BeatmapInfo.Length;
```

Comment:
Also this is completely incorrect, you should never use the beatmap online length for checking whether gameplay is finished (presuming that's what you want it for).

The last object of the beatmap is the one that will indicate that.

Reply:
A question: is `Beatmap.HitObjects` always ordered by their start time?
If not I'll have to manually order it

Reply:
Always ordered, yes.